### PR TITLE
Add nextPageLink/Count in ODataCollectionStart and write them to response in ODataJsonLightCollectionSerializer.cs

### PR DESCRIPTION
**Pull request by chinese007**
The change list here is the first step to fix issue(https://github.com/OData/odata.net/issues/140):

1. New properties have been introduced in ODataCollectionStart
2. If nextPageLink/Count is specified, the ODataJsonLightCollectionSerializer will write them into the response.

Another pull request will be made against WebAPI OData repository to have the issue fully addressed.

### CODE REVIEW

File: src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionSerializer.cs [hunk 1]
```diff
@@ -50,6 +50,20 @@ internal void WriteCollectionStart(ODataCollectionStart collectionStart, IEdmTyp
                 // "@odata.context":...
                 this.WriteContextUriProperty(ODataPayloadKind.Collection, () => ODataContextUrlInfo.Create(collectionStart.SerializationInfo, itemTypeReference));
 
+                // "@odata.count":...
```
Comment (congysu) [inline src/Microsoft.OData.Core/JsonLight/ODataJsonLightCollectionSerializer.cs]: @chinese007 Add some test cases. Also need to change the `WriteCollectionEnd`. De-serialize nextlink and count. Optional, depend on your user scenario, handle inline scenario

**Comment by lewischeng-ms** [comment]
Commits from @chinese007 : https://github.com/OData/odata.net/commit/5b03cc4f2c5329d18525661271967014025e5a9b and https://github.com/OData/odata.net/commit/ddc5eeaf656f79bb3a6cdf87baa190107ccdb359

**Comment by lewischeng-ms** [comment]
Final commits: https://github.com/OData/odata.net/commit/b3a8af816557fff4f5160967b762d1a982c7900d, https://github.com/OData/odata.net/commit/f2682791a375a2b83e60a04b47a0bb98fb92f90b and https://github.com/OData/odata.net/commit/7a604237405626f9c626c2e7452a67d764a79b52

**Comment by chinese007** [comment]
@lewischeng-ms Thanks very much for the fix. Do we know which build will contain this fix? Thanks!

**Comment by lewischeng-ms** [comment]
Hi @chinese007 , if you want the binaries immediately, please use our nightly builds (https://github.com/OData/odata.net#nightly-builds). Otherwise, please wait for the next release (v6.15).

**Pull request closed without merging**